### PR TITLE
Add a new post about adding built-in function

### PR DESCRIPTION
**Pull request by queenypingcap**

### CODE REVIEW

**Comment by queenypingcap** [comment]
@shenli @ngaut @c4pt0r PTAL.

File: _posts/2016-12-19-adding-built-in-function.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		```
+	2). Register the name and the implementation to [`builtin.Funcs`](https://github.com/pingcap/tidb/blob/master/evaluator/builtin.go#L43).
+  
+3. Add the Type Inference information to the [plan/typeinferer.go](https://github.com/pingcap/tidb/blob/master/plan/typeinferer.go) file. Add the type of the returned result of the function to `handleFuncCallExpr()` in the the [plan/typeinferer.go](https://github.com/pingcap/tidb/blob/master/plan/typeinferer.go) file and make sure the result is consistent with the result in MySQL. See [MySQL Const](https://github.com/pingcap/tidb/blob/master/mysql/type.go#L17) for the complete list of the type definition.
+4. Add a unit test case for the function to the [evaluator](https://github.com/pingcap/tidb/tree/master/evaluator) directory.
+4. Run the `make dev` command and make sure all the test cases can pass.
+
+### Example
+
+Take the [Pull Rest](https://github.com/pingcap/tidb/pull/2249) to add the `timediff()` function as an example:
```
Comment (shenli) [inline _posts/2016-12-19-adding-built-in-function.md]: Rest->Request

**Comment by shenli** [comment]
@coocood PTAL

**Comment by c4pt0r** [comment]
LGTM

**Comment by c4pt0r** [comment]
@shenli @siddontang PTAL

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**